src/classes/frAnswer.cls

```diff
@@ -0,0 +1,137 @@
+        String funraiseSupporterId = String.valueOf(request.get('supporterId'));
+        if(String.isNotBlank(funraiseSupporterId)) {
+            List<Contact> contacts = [SELECT Id from Contact WHERE fr_Id__c = :funraiseSupporterId];
+            if(contacts.isEmpty()) {
+                frUtil.logRelationshipError(frUtil.Entity.ANSWER, funraiseId, 
+                                            frUtil.Entity.SUPPORTER, funraiseSupporterId);
+            } else {
+                supporter = contacts.get(0);
+            }
+        }
+        
+        Opportunity opp = null;
+        String funraiseTransactionId = String.valueOf(request.get('donationId'));
+        if(String.isNotBlank(funraiseTransactionId)) {
+            List<Opportunity> transactions = [SELECT Id from Opportunity WHERE fr_Id__c = :funraiseTransactionId];
+            if(transactions.isEmpty()) {
+                frUtil.logRelationshipError(frUtil.Entity.ANSWER, funraiseId, 
+                                            frUtil.Entity.DONATION, funraiseTransactionId);
+            } else {
+                opp = transactions.get(0);
+            }
+        }
+        
+        Campaign page = null;
+        String funraisePageGoalId = String.valueOf(request.get('pageGoalId'));
+        if(String.isNotBlank(funraisePageGoalId)) {
+            List<Campaign> pages = [SELECT Id from Campaign WHERE fr_Id__c = :funraisePageGoalId];
+            if(pages.isEmpty()) {
+                frUtil.logRelationshipError(frUtil.Entity.ANSWER, funraiseId, 
+                                            frUtil.Entity.CAMPAIGN, funraisePageGoalId);
+            } else {
+                page = pages.get(0);
+            }
+        }
+        
+        Fundraising_Event_Registration__c registration = null;
+        String funraiseRegistrationId = String.valueOf(request.get('registrationId'));
+        if(String.isNotBlank(funraiseRegistrationId)) {
+            List<Fundraising_Event_Registration__c> registrations = [SELECT Id from Fundraising_Event_Registration__c WHERE fr_Id__c = :funraiseRegistrationId];
+            if(registrations.isEmpty()) {
+                frUtil.logRelationshipError(frUtil.Entity.ANSWER, funraiseId, 
+                                            frUtil.Entity.REGISTRATION, funraiseRegistrationId);
+            } else {
+                registration = registrations.get(0);
+            }
+        }
+        
+        Answer__c answer = new Answer__c(
+        	fr_Id__c = funraiseId,
+            Answer__c = frUtil.truncateToFieldLength(
+                Answer__c.Answer__c.getDescribe(), 
+                String.valueOf(request.get('answer'))
+            ),
+            Question__c = questions.get(0).Id,
+            Supporter__c = supporter != null ? supporter.Id : null,
+            Transaction__c = opp != null ? opp.Id : null,
+            Campaign_Page__c = page != null ? page.Id : null,
+			Registration__c = registration != null ? registration.Id : null
```

Comment:
The absolute most minor, only fix if there is other stuff we find, there is some whitespace weirdness here